docs/src/documentation/03-components/04-overlay/dialog/03-accessibility.mdx

```diff
@@ -0,0 +1,120 @@
+| triggerRef  | `React.RefObject<HTMLElement>`                          | References the element that triggered the dialog, allowing focus to return to it when the dialog closes.                        |
+
+### Automatic Accessibility Features
+
+The Dialog component automatically implements the following accessibility features:
+
+- `role="dialog"`: Identifies the element as a dialog to assistive technologies.
+
+- `aria-modal="true"`: Indicates that the dialog is modal, meaning it blocks interaction with page content.
+
+- `aria-labelledby`: Automatically associates the dialog with its title for screen readers using a generated ID.
+
+- `aria-describedby`: When a description is provided, it's automatically associated with the dialog using a generated ID.
+
+- The Dialog's animations respect the user's reduced motion preferences.
+
+### Focus Management
+
+- When opened, focus is automatically moved to the first focusable element within the dialog.
+
+- When closed, focus returns to the element that triggered the dialog (when `triggerRef` is provided).
+
+### Best practices
+
+- Always provide a descriptive `title` that clearly indicates the purpose of the dialog.
+
+- Use the `description` prop to provide additional context when needed, especially for complex interactions.
+
+- Ensure that primary and secondary actions have clear, descriptive labels that indicate their purpose.
+
+- Use semantic heading levels (`titleAs` prop) that fit within your page's heading hierarchy. For example, if your dialog appears within a page section that uses `h2`, consider using `titleAs="h3"` for proper document structure.
+
+- Test keyboard navigation within the dialog to ensure all interactive elements are accessible.
+
+- When creating a dialog trigger, help screen reader users understand the relationship between the trigger and the dialog:
+  1. Apply `aria-expanded="true"` to the trigger element when the dialog is open and `aria-expanded="false"` when it's closed.
+  2. Use `aria-controls` on the trigger element to associate it with the dialog via Dialog component's `id`.
+
+### Keyboard Navigation
+
+The Dialog component supports the following keyboard interactions:
+
+- **Escape** key closes the dialog
+- **Tab** key navigates through focusable elements within the dialog, with focus being contained within the dialog while it's open
+- **Enter/Space** keys activate buttons and other interactive elements within the dialog
+
+### Examples
+
+#### Basic Dialog with accessible title and semantic heading level
+
+```jsx
+<Dialog
+  title="Edit your profile information"
+  titleAs="h2"
+  description="Update your personal details to keep your account information current."
+  primaryAction={<Button type="critical">Save changes</Button>}
+  secondaryAction={<Button type="secondary">Cancel</Button>}
+  onClose={() => handleClose()}
```

Comment:
In the other example (1st) you had `triggerRef` prop, not sure if it was deleted on purpose

Reply:
Yeah, I removed it as it's used in the other example as well and there there is also the actual button and state values  so it felt better to use it only there properly.